### ISSUE TITLE
[1.3-beta.16] Error with cosmos:browserify: "Arguments to path.resolve must be strings"

### ISSUE THREAD

**Issue by servermeta** [meteor/meteor#6449]
Beta 16 is conflicting with flow-router:

```
mamluk@mamluk-ThinkPad-X1-Carbon-2nd:~/Projects/tmt$ meteor add kadira:flow-router
mamluk@mamluk-ThinkPad-X1-Carbon-2nd:~/Projects/tmt$ meteor
[[[[[ ~/Projects/tmt ]]]]]                    

=> Started proxy.                             
=> Started MongoDB.                           
=> Errors prevented startup:                  
   
   While processing files with cosmos:browserify (for target web.browser):
   packages/kadira:flow-router/client.browserify.js: Arguments to path.resolve
   must be strings
   
   Browserify options:
   >  undefined
   
=> Your application has errors. Waiting for file change.

```

**Comment by benjamn** [comment]
I can reproduce this, but I'm not yet sure what's causing it.

I also noticed that everything works if you install both [`cosmos:browserify`](https://github.com/elidoran/cosmos-browserify/) and [`kadira:flow-router`](https://github.com/kadirahq/flow-router/) as local packages.

Everything also works great if you change [this line](https://github.com/kadirahq/flow-router/blob/02adb1fbbdd3eb5d9c03464c5ea4646035882e44/package.js#L65) to `api.use("modules");`, so that `cosmos:browserify` is no longer used by `kadira:flow-router`… so that's probably the best answer once 1.3 is officially released.

**Comment by servermeta** [comment]
Great, thank you @benjamn , keep up the good work.

An issue has already been opened on flow router github.

What should us user do? report issues withe packages, at least of the more popular ones, or keep them for after 1.3 is out?

**Comment by markoshust** [comment]
Thanks @benjamn, you win the honey pot. I submitted a PR for https://github.com/kadirahq/flow-router/issues/552 with these updates. It obviously breaks for Meteor 1.2, so bumped minor version (this should really become 3.0.0 because it's a breaking update, but I have no clue what is going on with FlowRouter 3.0.

**Comment by guidobouman** [comment]
@benjamn I'm experiencing this issue as well, but adding `cosmos:browserify` didn't solve the issue in our case.

Any other thoughts to solve this issue for people rolling with 1.3-rc.2 already?

**Comment by area** [comment]
I'm using https://github.com/markoshust/flow-router/tree/feature/6449 explicitly with [mgp](https://github.com/DispatchMe/mgp).

**Comment by markoshust** [comment]
@area thanks. i just cloned out flow-router to packages/flow-router, made that changeset update, and made sure .meteor/packages was pointing at that new version.

**Comment by ffxsam** [comment]
Still an issue in RC2.

**Comment by sscaff1** [comment]
I'm running into this issue with 1.3 RC2 as well...

**Comment by gsabran** [comment]
I'm having the same issue, and it seems to touch other packages as well:
```
$ meteor
[[[[[ ~/project ]]]]]           

=> Started proxy.                             
=> Started MongoDB.                           
=> Errors prevented startup:                  
   
   While processing files with cosmos:browserify (for target web.browser):
   packages/kadira:flow-router/client.browserify.js: Arguments to path.resolve
   must be strings
   
   Browserify options:
   >  undefined
   packages/react-runtime-dev/react.browserify.js: Arguments to path.resolve
   must be strings
   
   Browserify options:
   >  undefined
   
   While processing files with cosmos:browserify (for target os.osx.x86_64):
   packages/react-runtime-dev/react.browserify.js: Arguments to path.resolve
   must be strings
   
   Browserify options:
   >  undefined
   
=> Your application has errors. Waiting for file change.
```

**Comment by cstrat** [comment]
I am seeing this issue too, however only after running `meteor reset`.
Prior to doing the mongo reset, everything worked 100% fine.
Even deployed on a server.

**Comment by markoshust** [comment]
@gsabran flow router uses browserify. check out recent issues, I have a fix for that. your other problem is that you are using meteor's react implementation -- this is a no-no with 1.3+. use npm's react.

**Comment by gsabran** [comment]
ah ok thanks for react. I thought it was what Meteor 1.3 was using...

**Comment by markoshust** [comment]
i did think meteor 1.3 was going to change their react implementation to use npm's react.. is this no longer the case?

**Comment by cstrat** [comment]
@gsabran to use react in 1.3 check out aronuda's post here:

https://voice.kadira.io/getting-started-with-meteor-1-3-and-react-15e071e41cd1

He makes a point:

> Do not use any Meteor packages that depend on react-runtime. That would load two versions of React, which would then cause a lot of other issues.
(That’s why we are using react-mounter instead of ReactLayout.)

**Comment by gsabran** [comment]
@cstrat thanks. I was using npm react and somehow react-runtime ended in my Meteor packages. To clarify: meteor 1.3 does use npm's react.
I hope they had a way to reconcile meteor and npm dependencies for any use case, instead of having to be careful of what gets loaded where (either directly in npm, which is cool, or in meteor packages, which is not that cool when it's mainly a npm thing) Thanks! and I'll stop commenting on something unrelated to the open issue

**Comment by xcv58** [comment]
Same for rc2, two packges have this issue:
```
packages/react-runtime-dev/react.browserify.js: Arguments to path.resolve must be strings
packages/xcv58:auth0-lock/auth0-lock.browserify.js: Arguments to path.resolve must be strings
```

**Comment by cstrat** [comment]
I found a [quick fix](https://forums.meteor.com/t/quick-fix-for-cosmos-browserify-error-on-1-3-rc2/19633/1) for this issue, not sure if it will work for everyone. Just downgrade to beta 12 and then back to rc2 and it seems to fix things for me.

**Comment by xcv58** [comment]
@cstrat If you run meteor reset, the same problem appears.

**Comment by cstrat** [comment]
@xcv58 yeah after you run reset, downgrade to b12, then back to rc2.

**Comment by benjamn** [comment]
Fixed in `1.3-rc.3`.

**Comment by scheung38** [comment]
Does kadira:flow-router work with 1.3.1 then?
